src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -75,7 +92,11 @@ public Double averageSumOdd(List<Integer> numbers) {
      **/
     public List<People> workablePeople(int fromAge, int femaleToAge,
                                        int maleToAge, List<People> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getAge() >= fromAge
+                        && (p.getSex() == People.Sex.MAN && p.getAge() <= maleToAge)
+                        || (p.getSex() == People.Sex.WOMEN && p.getAge() <= femaleToAge))
```

Comment:
```suggestion
                .filter(p -> p.getAge() >= fromAge
                        && (p.getSex() == People.Sex.MAN && p.getAge() <= maleToAge
                        || p.getSex() == People.Sex.WOMEN && p.getAge() <= femaleToAge))
```
don't forget that `&&` is evaluated before `||`, and you need to wrap the whole right part so that it is evaluated all together